src/main/java/com/fdmgroup/forex/models/Trade.java

```diff
@@ -0,0 +1,88 @@
+
+package com.fdmgroup.forex.models;
+
+import java.util.*;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.springframework.data.annotation.Immutable;
+
+import com.fdmgroup.forex.models.composites.TradeId;
+
+import jakarta.persistence.*;
+
+@Entity
+@Immutable
+@Table(name = "trades")
+@IdClass(TradeId.class)
+public class Trade {
+
+    @Id
```

Comment:
do we need auto generate for this? or should I generate it when i create?

Reply:
ok i think i generate it because we need both id be the same

Reply:
Correct - you'll need to pass it in when creating both trade records